File: server/runningServer/src/main/java/com/server/running/user/service/UserService.java
package com.server.running.user.service;

import java.util.List;

import com.server.running.user.dto.Friend;
import com.server.running.user.dto.User;

public interface UserService {
	// 회원가입
	public boolean signup(User user);
	
	// 로그인
	public User login(User user);
	
	// 회원 정보 수정
	public User updateUser(User user);
	
	// 회원 탈퇴
	public boolean deleteUser(User user);

	// 러닝 데이터 조회
	public User findRunning(int uid);
	
	// 아이디 중복 체크 요청
	public boolean overlap(User user);
	
	// 친구 검색
	public List<String> findFriends(String email);

	// 친구 추가
	public Boolean addFriend(Friend friend);
}

File: client/app/src/main/java/com/example/runningapplication/service/UserService.kt
package com.example.runningapplication.service

import com.example.runningapplication.data.model.Challenge
import com.example.runningapplication.data.model.User
import retrofit2.http.*
import retrofit2.Call

interface UserService {

//    @FormUrlEncoded
    @POST("/signup.run")
    fun signUp(
        @Body parameters: HashMap<String,Any>
    ):Call<Boolean>

    @POST("/login.run")
    fun login(
        @Body parameters: HashMap<String,Any>
    ):Call<User>

    @POST("/overlap.run")
    fun emailCheck(
        @Body parameters: HashMap<String,Any>
    ):Call<Boolean>


    @GET("/findAllTeam.run")
    fun findTeam(
        @Query("uid") uid:Int
    ):Call<List<Challenge>>

    @GET("/findFriends.run")
    fun findFriends(
        @Query("email") email: String
    ):Call<List<String>>

    @POST("/friendAdd.run")
    fun friendAdd(
        @Body param: HashMap<String, Any>
    ):Call<Boolean>

    @POST("/joinTeam.run")
    fun joinTeam(
        @Body param: HashMap<String, Any>
    ):Call<Boolean>
}